File: public/usage-examples/interface/set_interface_element_color-1-example.py
from splashkit import *


open_window("Interface Element Contrast", 700, 420)

element_color = rgb_color(70, 160, 220)
contrast = 0.0

while not quit_requested():
    process_events()

    # Change the interface contrast using number keys
    if key_typed(KeyCode.num_1_key):
        contrast = 0.0

    if key_typed(KeyCode.num_2_key):
        contrast = 1.0

    # Apply the selected color and contrast to the interface
    set_interface_element_color(element_color, contrast)

    clear_screen(color_white())

    draw_text_font_as_string(
        "Press 1 for minimum contrast or 2 for maximum contrast",
        color_black(),
        "Arial",
        20,
        90,
        80
    )

    draw_text_font_as_string(
        "Current contrast: " + str(contrast),
        color_black(),
        "Arial",
        20,
        240,
        130
    )

    button_at_position(
        "Interface Button",
        rectangle_from(230, 200, 240, 60)
    )

    draw_interface()
    refresh_screen_with_target_fps(60)

close_all_windows()

File: public/usage-examples/interface/set_interface_element_color-1-example.cpp
#include "splashkit.h"

int main()
{
    open_window("Interface Element Contrast", 700, 420);

    color element_color = rgb_color(70, 160, 220);
    float contrast = 0.0f;

    while (!quit_requested())
    {
        process_events();

        // Change the interface contrast using number keys
        if (key_typed(NUM_1_KEY))
        {
            contrast = 0.0f;
        }

        if (key_typed(NUM_2_KEY))
        {
            contrast = 1.0f;
        }

        // Apply the selected color and contrast to the interface
        set_interface_element_color(element_color, contrast);

        clear_screen(color_white());

        draw_text(
            "Press 1 for minimum contrast or 2 for maximum contrast",
            color_black(),
            90,
            80
        );

        draw_text(
            "Current contrast: " + std::to_string(contrast),
            color_black(),
            240,
            130
        );

        button(
            "Interface Button",
            rectangle_from(230, 200, 240, 60)
        );

        draw_interface();
        refresh_screen(60);
    }

    close_all_windows();

    return 0;
}